corehq/apps/receiverwrapper/util.py

```diff
@@ -179,3 +181,32 @@ def determine_authtype(request):
         return request.GET['authtype']
 
     return determine_authtype_from_request(request)
+
+
+def from_demo_user(form_json):
+    """
+    Whether the form is submitted by demo_user
+    """
+    from corehq.apps.users.util import DEMO_USER_ID
+    try:
+        # require new-style meta/userID (reject Meta/chw_id)
+        if form_json['meta']['userID'] == DEMO_USER_ID:
+            return True
+    except (KeyError, ValueError):
+        return False
+
+
+def should_ignore_submission(request):
+    """
+    If submission request.GET has `submit_mode=demo` and submitting user is not demo_user,
+    the submissions should be ignored
+    """
+    if not request.GET.get('submit_mode') == 'demo':
+        return False
+
+    instance, _ = couchforms.get_instance_and_attachment(request)
+    form_json = convert_xform_to_json(instance)
+    if not from_demo_user(form_json):
+        return True
+    else:
+        return False
```

Comment:
nit: inverting this if makes it easier to read: `if from_demo_user return False else return True`